### ISSUE TITLE
Comet 2024.01 crashes

### ISSUE THREAD

**Issue by cbielow** [UWPR/Comet#63]
another test case in OpenMS crashes with
```
 Comet version "2024.01 rev. 0"

 Search start:  07/04/2024, 10:58:53 AM
 - Input file: C:/dev/openms/src/tests/topp/THIRDPARTY/spectra_comet.mzML
   - Load spectra: 3
     - Search progress:
 Error in StorePeptides.  stored twice:  1 0.000000, 0 0.000000
```
The error is here: [CometSearch.cpp#L4027](https://github.com/UWPR/Comet/blob/master/CometSearch/CometSearch.cpp#L4027), and it seems this should not have happened :/

Previous Comet versions run fine.

I will attach the [param.txt](https://github.com/user-attachments/files/16096165/param.txt) here.
The mzML can be downloaded [here](https://github.com/OpenMS/OpenMS/blob/develop/src/tests/topp/THIRDPARTY/spectra_comet.mzML), the FASTA [here](https://github.com/OpenMS/OpenMS/blob/develop/src/tests/topp/THIRDPARTY/proteins.fasta)

(probably unrelated to #62)

**Comment by jke000** [comment]
I have the fix for this also which will be committed soon. Thanks for reporting this one also.

**Comment by jke000** [comment]
This is addressed by commit e5cf236.
